common/src/main/java/com/revenuecat/purchases/common/verification/SigningManager.kt

```diff
@@ -77,11 +79,13 @@ class SigningManager(
             }
             is Result.Success -> {
                 val intermediateKeyVerifier = result.value
-                val decodedNonce = nonce?.let { Base64.decode(it, Base64.DEFAULT) } ?: byteArrayOf()
-                val requestTimeBytes = requestTime.toByteArray()
-                val eTagBytes = eTag?.toByteArray() ?: byteArrayOf()
-                val payloadBytes = body?.toByteArray() ?: byteArrayOf()
-                val messageToVerify = signature.salt + decodedNonce + requestTimeBytes + eTagBytes + payloadBytes
+                val messageToVerify = signature.salt +
+                    apiKey.toByteArray() +
+                    (nonce?.let { Base64.decode(it, Base64.DEFAULT) } ?: byteArrayOf()) +
+                    Uri.decode(urlPath).toByteArray() +
+                    requestTime.toByteArray() +
+                    (eTag?.toByteArray() ?: byteArrayOf()) +
+                    (body?.toByteArray() ?: byteArrayOf())
```

Comment:
I was going to suggest extracting this into a private function but it might have too many parameters?

Reply:
Yeah, I wasn't in love this... In the end, I changed it to imitate what we do in iOS. Basically I created a new data holder for all this so we can move the creation of the signature to verify there. Still long, but provides a better abstraction IMO.